### DISCUSSION TITLE
Difference between run a Docker imager or run Keycloak in production?

### DISCUSSION THREAD

**Discussion by tiagogalvao7** [keycloak/keycloak#28487] [Q&A]
Hello goodnight, I will use Keycloak in a project that requires Keycloak for OpenID Connect Provider solution. I'm a little bit confused about where I need to start, and my question is: what is the difference between run a Docker image for Keycloak, and run Keycloak in production?

**Comment by codespearhead** [comment]
The Docker (or rather, the OCI-compliant container) image is the same.

Running it in production mode optimizes configurations for security and performance. This includes enforcing essential features like mandatory use of TLS/SSL for secure communication and automatic caching mechanisms for improved speed.

**Reply by tiagogalvao7** [reply]
So, it's supposed to work with Keycloak in production mode?

**Reply by codespearhead** [reply]
I'm not sure I understand your question.

If you're asking whether you should always run software in production mode, then the answer is yes, unless you're working on the codebase or exploring its functionalities in an air-gapped environment.
